### PR TITLE
[v2.6] Refactor RKE2/K3s provisioning tests

### PR DESCRIPTION
**Pull request by markusewalker**
## Problem

Recently, we refactored the hosted provisioning tests to do two of the following:

- Rework redundant provisioning code into a private method
- Separate provisioning test and additional test cases into their own, unique test functions

This greatly helps moving forward for internal CI/CD pipeline and overall helps to better organize our ever-growing test framework. As such, the same approach has been made to the RKE2/K3s provisioning tests.
 
## Solution

Like with the hosted provisioning tests, the redundant provisioning code for node provider and custom clusters has been put into a single private method. The now separated unique test functions reference this private method and only keep unique code in their respective functions.

Additionally, I added two new Windows test for specifically the DynamicInput function. This is because before there were no Windows test cases in the DynamicInput, only the static counterpart had Windows tests.

Lastly, I added a check in the custom cluster tests to see if the K8s version is less than 1.25.0. This is because PSPs are officially removed in K8s 1.25, so we need to update our tests to account for this if a cluster version is > 1.25.0.
 
### Manual Testing

Locally tested static and dynamic input functions for both node provider and custom cluster.

### Automated Testing

Jenkins runs will be provided offline to reviewers.

### CODE REVIEW

File: tests/v2/validation/provisioning/rke2/provisioning_node_driver_test.go
Comment (igomez06) [inline tests/v2/validation/provisioning/rke2/provisioning_node_driver_test.go:343]: Since we're removing this test I still think we should add this check to the main provisioning test. Checking the status pods.

**Review by igomez06 — COMMENTED** [review]
Looks great just one little suggestion/comment.

File: tests/v2/validation/provisioning/rke2/provisioning_node_driver_test.go
Comment (markusewalker) [reply tests/v2/validation/provisioning/rke2/provisioning_node_driver_test.go:343]: That's a good idea, I went ahead and put it back here and in the 2.7 forwardport.

**Review by thaneunsoo — APPROVED** [review]
lgtm

**Review by igomez06 — APPROVED** [review]
Looks good! I would like to hold off on merging this until we have the code coverage branches merged too, but that will be quick.

**Comment by markusewalker** [comment]
Rebased changes based upon latest code coverage PR change. Once drone passes, we're good to merge.

**Review by igomez06 — APPROVED** [review]
LGTM

**Review by slickwarren — APPROVED** [review]
LGTM

**Pull request merged**
